src/main/java/org/openstreetmap/atlas/generator/sharding/OverpassClient.java

```diff
@@ -59,6 +60,16 @@ public static String buildQuery(final String type, final String key, final Strin
                 + END_QUERY;
     }
 
+    private static void checkForTooMuchData(final Document document)
+    {
+        if (document.getDocumentElement().getElementsByTagName("remark").item(0) != null
+                && document.getDocumentElement().getElementsByTagName("remark").item(0)
+                        .getTextContent().contains("runtime error: Query ran out of memory"))
```

Comment:
Is it possible to catch any type of error here? That would make the verification tool more robust, right?

Reply:
Since it's not a response code that we read but instead the contents of the returned xml document, I'd have to see multiple errors and look for common keywords to catch for. I suppose we could fail if we just see the word error in the remark tag, but for that case we would have to write a more general exception message.